src/core/Akka.Cluster/ClusterDaemon.cs

```diff
@@ -895,7 +895,7 @@ private void CreateChildren(Cluster cluster)
             _cluster = cluster;
             _coreSupervisor = Context.ActorOf(Props.Create<ClusterCoreSupervisor>(), "core");
 
-            Context.ActorOf(ClusterHeartbeatReceiver.Props(() => _cluster), "heartbeatReceiver");
```

Comment:
Avoid creating a delegate for the `Lazy<Cluster>` that we're removing from the `HeartbeatReceiver`